src/ua/artcode/udiary/tests/TestMainController.java

```diff
@@ -0,0 +1,321 @@
+package ua.artcode.udiary.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import ua.artcode.udiary.controller.MainController;
+import ua.artcode.udiary.controller.MainControllerImpl;
+import ua.artcode.udiary.dao.AppDataContainer;
+import ua.artcode.udiary.dao.DairyDaoImpl;
+import ua.artcode.udiary.dao.RecordDaoImpl;
+import ua.artcode.udiary.dao.UserDaoImpl;
+import ua.artcode.udiary.exception.AppException;
+import ua.artcode.udiary.model.*;
+
+public class TestMainController {
+
+    private MainController mainController;
+
+    @Before
+    public void prepareTest() {
+        AppDataContainer appDataContainer = new AppDataContainer();
+        mainController = new MainControllerImpl(
+                new RecordDaoImpl(appDataContainer),
+                new DairyDaoImpl(appDataContainer),
+                new UserDaoImpl(appDataContainer)
+        );
+    }
+
+    @Test
+    public void addUser() {
```

Comment:
one test = several asserts, i mean one test = one test logic, do not put many asserts in one test

Reply:
Fixed. Please, check new version in the next push.